comid/measurement_test.go

```diff
@@ -112,3 +123,227 @@ func TestMeasurement_NewUUIDMeasurement_bad_uuid(t *testing.T) {
 
 	assert.Nil(t, tv.SetUUID(nonRFC4122UUID))
 }
+
+var (
+	goldenMKey  uint64 = 100
+	goldenMKey1 uint64 = 120
+)
+
+func TestUIntMkey_NotSet(t *testing.T) {
+	mkey := &Mkey{}
+	expectedErr := "unknown measurement key type: <nil>"
+	err := mkey.Valid()
+	require.NotNil(t, err)
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestUIntMkey_MarshalCBOR_ok(t *testing.T) {
+	tvs := []struct {
+		mkey     uint64
+		expected []byte
+	}{
+		{
+			mkey:     goldenMKey,
+			expected: ([]byte{0x18, 0x64}),
+		},
+		{
+			mkey:     goldenMKey1,
+			expected: ([]byte{0x18, 0x78}),
+		},
+	}
+
+	for _, tv := range tvs {
+		meas := NewUintMeasurement(tv.mkey)
+		require.NotNil(t, meas)
+
+		actual, err := meas.Key.MarshalCBOR()
+
+		fmt.Printf("CBOR: %x\n", actual)
+
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+	}
+}
+func TestUIntMkey_MarshalCBOR_notok(t *testing.T) {
+	tvs := []struct {
+		mkey     uint64
+		expected string
+	}{
+		{
+			mkey:     0,
+			expected: "unknown measurement key type: <nil>",
+		},
+	}
+
+	for _, tv := range tvs {
+		mkey := &Mkey{}
+		actual, err := mkey.MarshalCBOR()
+
+		fmt.Printf("CBOR: %x\n", actual)
+
+		assert.NotNil(t, err)
+		assert.EqualError(t, err, tv.expected)
+	}
+}
+
+func TestUIntMkey_UnmarshalCBOR_ok(t *testing.T) {
+	tvs := []struct {
+		input    []byte
+		expected uint64
+	}{
+		{
+			input:    []byte{0x18, 0x64},
+			expected: goldenMKey,
+		},
+		{
+			input:    []byte{0x18, 0x78},
+			expected: goldenMKey1,
+		},
+	}
+
+	for _, tv := range tvs {
+		mKey := &Mkey{}
+		require.NotNil(t, mKey)
+
+		err := mKey.UnmarshalCBOR(tv.input)
+		assert.Nil(t, err)
+		actual, err := mKey.GetKeyUint()
+		assert.Nil(t, err)
+		assert.Equal(t, tv.expected, actual)
+	}
+}
+
+func TestUIntMkey_UnmarshalCBOR_notok(t *testing.T) {
+	tvs := []struct {
+		input    []byte
+		expected string
+	}{
+		{
+			input:    []byte("Garbage"),
+			expected: "unexpected EOF",
+		},
+		{
+			input:    []byte(`{"type":"uint","value":123.456}`),
+			expected: "unexpected EOF",
+		},
```

Comment:
since this is negative test for the CBOR decoder we should provide invalid CBOR here rather than invalid JSON, no?